app/models/concerns/redis_caching.rb

```diff
@@ -19,8 +19,12 @@ def get_cached(key)
     end
 
     def set_cached(key, data)
-      @redis.set(key, data.to_json)
-      @redis.expire(key, @redis_ttl)
+      if data
+        @redis.set(key, data.to_json)
+        @redis.expire(key, @redis_ttl)
+      else
+        Rails.logger.debug('Nil data for redis caching')
```

Comment:
I'm not sure we'll ever see `debug` level logging in staging/prod like environments if we are trying to chase this down. info might be better. And I would change the message to something like "Attempted to set nil data in redis cache". But not a huge deal, the meaning is clear enough.